python/tk_multi_workfiles/file_list/file_widget.py

```diff
@@ -112,6 +128,15 @@ def set_thumbnail(self, thumb):
             thumb = QtGui.QPixmap(":/tk-multi-workfiles2/thumb_empty.png")
         self._ui.thumbnail.setPixmap(thumb)
 
+    def set_badge(self, badge):
+        if isinstance(badge, QtGui.QPixmap):
+            self._badge_icon.setPixmap(badge)
+            self._badge_icon.setVisible(True)
+        else:
+            # If we don't have a QPixmap badge, clear and hide the badge.
```

Comment:
Could you elaborate why we wouldn't have a QPixmap image? Should we raise an error or log something?

Reply:
Improved the comment.  This could be None or QColor, if there is no badge or the badge hasn't been generated yet.  In either case, we should not show the badge.  Improved comment should help clarify.